src/common/bloom_filter.hpp

```diff
@@ -483,10 +470,10 @@ class bloom_filter
 
   inline bloom_type hash_ap(uint32_t val, bloom_type hash) const
   {
-    hash ^=    (hash <<  7) ^  ((val & 0xff000000) >> 24) * (hash >> 3);
-    hash ^= (~((hash << 11) + (((val & 0xff0000) >> 16) ^ (hash >> 5))));
-    hash ^=    (hash <<  7) ^  ((val & 0xff00) >> 8) * (hash >> 3);
-    hash ^= (~((hash << 11) + (((val & 0xff)) ^ (hash >> 5))));
+    hash ^=    (hash <<  7) ^  ((val & 0xff)) * (hash >> 3);
+    hash ^= (~((hash << 11) + (((val & 0xff00) >> 8) ^ (hash >> 5))));
+    hash ^=    (hash <<  7) ^  ((val & 0xff0000) >> 16) * (hash >> 3);
+    hash ^= (~((hash << 11) + (((val & 0xff000000) >> 24) ^ (hash >> 5))));
```

Comment:
@uweigand what the fpp before and after this change in our unit tests?

> Changing the uint32_t variants of the bloom filter implementation to match the behavior of the POD implementation on little-endian systems (instead of big-endian), which results in the desired density.

this change is not backward compatible. as we do persist the `bit_table_` to disk in the cache tier 
see
https://github.com/ceph/ceph/blob/fa1ee5b4f63a44658d7879c09551ff09357345e9/src/osd/HitSet.h#L324
and https://github.com/ceph/ceph/blob/fa1ee5b4f63a44658d7879c09551ff09357345e9/src/osd/HitSet.h#L436

i am wondering if 

1. we can bump the version of `BloomHitSet`
1. use the old hash function for the BloomHitSet v1
1. use the new hash function for the BloomHitSet v2

that'd ideal.

@liewegas @uweigand what do you think?